sql/lease.go

```diff
@@ -272,8 +278,8 @@ func (s LeaseStore) Publish(tableID ID, update func(*TableDescriptor) error) err
 			return txn.CommitInBatch(b)
 		})
 
-		if err != errLeaseVersionChanged {
-			return err
+		if _, ok := pErr.GoError().(*errLeaseVersionChanged); !ok {
```

Comment:
I've ended up adding more messages to error.proto to be able to convert between go Errors and  proto messages. Added a TODO to revisit the design.